src/data/components/groupaddbutton.js

```diff
@@ -21,5 +37,46 @@ const GroupAddButton = React.createClass({
   handleClick: function(event) {
     event.stopPropagation();
     this.props.onClick();
-  }
+  },
+
+  handleGroupDragOver: function(event) {
+    event.stopPropagation();
+    event.preventDefault();
+  },
+
+  handleDragEnter: function(event) {
+    event.stopPropagation();
+    event.preventDefault();
+
+    this._dragDropCounter++;
+    this.setState({draggingOver: true});
+  },
+
+  handleDragLeave: function(event) {
+    event.stopPropagation();
+    event.preventDefault();
+
+    this._dragDropCounter--;
+    if (this._dragDropCounter === 0) {
+      this.setState({draggingOver: false});
+    }
+  },
+
+  handleDrop: function(event) {
+    event.stopPropagation();
+
+    this.setState({draggingOver: false});
+    this._dragDropCounter = 0;
+
+    var sourceGroup = event.dataTransfer.getData("tab/group");
+    var tabIndex = event.dataTransfer.getData("tab/index");
+
+    this.props.onDrop(
+      sourceGroup,
+      tabIndex
+    );
+  },
+
+  _dragDropCounter: 0
```

Comment:
This should be inside the state. Actually, the counter should be the only thing in the state since you do not really need `draggingOver`. In line 15, you could simply assign `this.state.dragDropCounter == 0`.